### ISSUE TITLE
Version control device drivers instead of only sharing at release?

### ISSUE THREAD

**Issue by talonchandler** [mehta-lab/recOrder#152]
Yuming wants to try the background correction fixes on `main`, but he was unable to perform a calibration because he does not have the latest device drivers (v1.1.0 and the new .csv format introduced in #127). We were planning to only distribute the device drivers on each release, but this means that a keen user will be unable to run from `main`. 

Can we version control the device drivers (<1 MB  total) so that users can run from `main`?

@ieivanov @mattersoflight

**Comment by talonchandler** [comment]
For now, I've asked Yuming to work from the `bkg-correction-fixes` branch instead of `main` while we figure out if we'd like to support users on `main`.

**Comment by talonchandler** [comment]
I think we're trying to answer the question: what's the best way to get new features out to users fast? 

1) Support users on `main`? This is potentially a lot of work.
2) Support users on feature-specific branches? Intermediate amount of work. In this case Yuming specifically needs this feature, so I think it makes sense to support this branch. 
3) Ask users to wait until the next release? This is the least work for us, but requires the most waiting for users.

**Comment by ieivanov** [comment]
I think this will be a problem only until we upload the Meadowlark device adapter to the MM GitHub repo. At that point, users will be able to install the latest MM nightly build and run recOrder from the main branch.

In the meantime, I think we can fall back to Option 3 or distribute device adapters if users get in touch with us - which I think will be the most straightforward solution here.

**Comment by talonchandler** [comment]
Thanks Ivan, I agree that this issue disappears when the relevant drivers are distributed with the MM nightly build. 

Will `usbdrvd64.dll` be distributed via MM release, too? And are we planning to phase out the distribution of `mmgr_dal_MeadowlarkLC.csv` now that the Schnoor fit code has merged? I'm in support of "Yes" to both so that we don't have to distribute any files with our releases.

**Comment by ieivanov** [comment]
We'll instruct users to copy the `usbdrvd64.dll` from the MLO USB stick to the MM directory during first-time installation. This file will not change between device adapter versions, and we will not need to distribute it (I also don't think we have the right to).

The Schnoor fit currently relies on the `mmgr_dal_MeadowlarkLC.csv`. The data in the file is fit at each class init. We can easily save the fit parameters (6 floats total, I thinks) and recall them if the file does not exist.

**Comment by talonchandler** [comment]
All sounds good. Thanks Ivan. Closing.

**Comment by mattersoflight** [comment]
👍🏼 It is indeed best that the drivers we depend on are distributed via  Micro-Manager. In the meantime, it is useful to host all related files together for clean distribution. Github releases do [allow packaging of binaries](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository). 

Since we are thinking of release candidates for managing dependencies on waveOrder, we can use these releases to package binaries.

**Comment by talonchandler** [comment]
Sounds good. 

Just to clarify, I am currently distributing the device drivers via GitHub's "Releases" where I've added the binaries (and the single `.csv` as a `.zip`. I also link to these files in the [microscope installation guide](https://github.com/mehta-lab/recOrder/blob/main/docs/microscope-installation-guide.md) that I used to walk Yuming through the installation. I will keep these files up to date until the drivers are included in an MM release.
